### PR TITLE
fix NftMintButton mint without properties

### PR DESCRIPTION
**Pull request by dlaxcess**
Fix NftMintButton when minting in medias in wordpress (minting without properties)

### CODE REVIEW

File: common/lib/nft/storage/knft-ipfs.ts [hunk 1]
```diff
@@ -22,7 +22,7 @@ const nftIpfsJson = async (
   ipfs = "",
   address = "",
   image = "",
-  properties: properties = {},
+  properties: Properties = {},
```
Comment (zapaz) [inline common/lib/nft/storage/knft-ipfs.ts:25]: put new param in last position , i.e. properties after metadata

Comment (zapaz) [reply common/lib/nft/storage/knft-ipfs.ts:25]: line 39   `if (Object.keys(properties)?.length > 0) json.properties = properties;` =>  `if (Object.keys(properties).length > 0) json.properties = properties;`

File: svelte/components/Media/MediaPreview.svelte [hunk 2]
```diff
@@ -17,8 +17,16 @@
...
 
   let open = false;
-  let metadatas: NftMetadata;
-  $: metadatas = nft?.metadata as NftMetadata;
+  let metadatas: NftMetadata | undefined;
+  $: metadatas = nft?.metadata;
+
+  const handleClose = () => (open = false);
+  const handlemetadatas = () =>
+    !metadatas?.animation_url
```
Comment (zapaz) [inline svelte/components/Media/MediaPreview.svelte:25]: to be refactorized  (at least one bug inside)

File: svelte/components/Media/MediaPreview.svelte [hunk 3]
```diff
@@ -30,12 +38,8 @@
...
         ? 'no-zoom-hover'
         : 'zoom-hover'}"
-      on:click={() =>
-        !metadatas?.animation_url
-          ? (open = true)
-          : $toPlayTokenID !== nft.tokenID
-          ? ($toPlayTokenID = nft.tokenID)
-          : ($toPlayTokenID = "")}
+      on:click={handlemetadatas}
+      on:keydown={handlemetadatas}
```
Comment (zapaz) [inline svelte/components/Media/MediaPreview.svelte:42]: do the same for each on:click (i.e. add on key event, not this one in any case)

File: svelte/components/Nft/NftMintButton.svelte [hunk 4]
```diff
@@ -70,7 +70,7 @@
 
       minting = 2;
 
-      const ipfsJson = await nftIpfsJson(alt, alt, ipfsImage, signerAddress, src, null, metadata);
+      const ipfsJson = await nftIpfsJson(alt, alt, ipfsImage, signerAddress, src, {}, metadata);
```
Comment (zapaz) [inline svelte/components/Nft/NftMintButton.svelte:73]: to put as last param => but as same as default value : not needed

**Review by zapaz — CHANGES_REQUESTED** [review]

File: svelte/components/Media/MediaPreview.svelte (hunk 2 above)
Comment (dlaxcess) [reply svelte/components/Media/MediaPreview.svelte:25]: can you specify witch bug ?

File: svelte/components/Global/Content.svelte [hunk 5]
```diff
@@ -41,11 +41,11 @@
 </script>
 
 <!-- {tokenID} -->
-<div on:click={(evt) => handleClick(evt)}>
+<div on:click={(evt) => handleClick(evt)} on:keydown={(evt) => handleClick(evt)}>
```
Comment (zapaz) [inline svelte/components/Global/Content.svelte:44]: ` on:click={handleClick}`  should be ok ?

File: svelte/components/Media/MediaPreview.svelte (hunk 2 above)
Comment (zapaz) [inline svelte/components/Media/MediaPreview.svelte:25]: refacto !

**Review by zapaz — CHANGES_REQUESTED** [review]

**Pull request merged**